### PR TITLE
fix: npu_rms_norm does not support gamma=None, use all-ones tensor in…

### LINKED ISSUE #14590: RMSNorm crashes on NPU when elementwise_affine=False (weight=None): npu_rms_norm requires a real gamma tensor
### Describe the bug

`diffusers.models.normalization.RMSNorm` crashes on Ascend NPU when constructed
with `elementwise_affine=False` (i.e. `self.weight is None`).

The NPU branch of `RMSNorm.forward` passes `self.weight` straight into the
fused CANN op `torch_npu.npu_rms_norm`, but that op's schema declares `gamma`
as a required `Tensor` (not `Optional`), so a `None` weight is invalid and the
forward errors out. The non-NPU `else` branch handles `weight=None` correctly.

### Reproduction

```python
import torch
from diffusers.models.normalization import RMSNorm

# LTX-2 style block norm: no learnable affine => weight is None
norm = RMSNorm(dim=4096, eps=1e-6, elementwise_affine=False).npu().to(torch.bfloat16)
x = torch.randn(1, 256, 4096, device="npu", dtype=torch.bfloat16)
out = norm(x)   # raises on NPU
```

(Note: this needs an actual Ascend NPU device; on a non-NPU build it fails earlier
with a backend-availability error rather than the gamma=None error.)

### Logs

```shell
On Ascend NPU the call to `torch_npu.npu_rms_norm(hidden_states, None, eps)` violates
the op schema and the forward aborts. The op's schema (from `torch_npu`) is:


npu::npu_rms_norm(Tensor input, Tensor gamma, float epsilon=1e-6) -> (Tensor, Tensor)


`gamma` is a required `Tensor`, while `RMSNorm` can legitimately have
`self.weight = None` (when `elementwise_affine=False`, i.e. gamma == 1, no affine).
```

### System Info

- 🤗 Diffusers version: 0.40.0.dev0
- Platform: Linux-5.15.0-119-generic-x86_64-with-glibc2.39
- Running on Google Colab?: No
- Python version: 3.11.15
- PyTorch version (GPU?): 2.9.0+cpu (False)
- Huggingface_hub version: 1.27.0
- Transformers version: 5.3.0
- Accelerate version: 1.10.1
- PEFT version: 0.18.0
- bitsandbytes version: 0.49.2
- optimum-quanto version: 0.2.7
- Safetensors version: 0.8.0
- xFormers version: not installed
- Accelerator: Ascend NPU (torch_npu 2.9.0.dev20260207)
- Using GPU in script?: Yes (Ascend NPU via `torch_npu`, `is_torch_npu_available()==True`)
- Using distributed or parallel set-up in script?: Yes (accelerate multi-NPU)

### Who can help?

_No response_

### PR DESCRIPTION
**Pull request by xucqX**
fixes: #14590 
## What does this PR do?

Fixes a crash in `RMSNorm.forward` on Ascend NPU when the layer is built with
`elementwise_affine=False` (i.e. `self.weight is None`).

### Context

`RMSNorm` supports `elementwise_affine=False`, which sets `self.weight = None`
(no learnable gamma, mathematically gamma == 1). The non-NPU `else` branch
handles this fine, but the NPU branch explicitly calls the fused CANN op

```
torch_npu.npu_rms_norm(hidden_states, self.weight, epsilon=self.eps)
```

and passes `self.weight` straight through. `npu_rms_norm` does **not** accept
`gamma=None` (it requires a real tensor), so any model that uses weightless
RMSNorm blocks crashes on Ascend the moment forward hits one of them.

Concrete in-repo trigger: `LTX2VideoTransformerBlock`
(`src/diffusers/models/transformers/transformer_ltx2.py`) constructs its block
norms (`norm1/norm2/norm3`, the audio variants, and the a2v/v2a cross-attn
norms) with `elementwise_affine=False`, so LTX-2 is unusable on NPU without
this fix.

### The fix

When `self.weight is None`, substitute an all-ones tensor of the same shape,
dtype and device:

```python
weight = self.weight
if weight is None:
    weight = torch.ones(self.dim, device=hidden_states.device, dtype=hidden_states.dtype)
if weight.dtype in [torch.float16, torch.bfloat16]:
    hidden_states = hidden_states.to(weight.dtype)
hidden_states = torch_npu.npu_rms_norm(hidden_states, weight, epsilon=self.eps)[0]
```

Multiplying by an all-ones tensor is identical to "no affine" (gamma=1), so
this preserves the original semantics while giving the CANN op a valid tensor.
The CPU/GPU `else` branch and bias handling are unchanged.

No new dependencies.

## Self-review notes (AI-assisted)

- **Math equivalence:** `ones` ≡ gamma=1 ≡ the "no affine" semantics the
  `else` branch already implements for the `weight=None` case. No behaviour
  change on CPU/GPU.
- **Alternative considered:** fall back to the `else` (decomposed) branch on
  NPU when `weight is None`. Rejected — the explicit NPU branch exists
  precisely to get the fused CANN kernel in eager mode; the ones-shim keeps
  the fast path rather than degrading to a multi-launch decomposition.
- **dtype/device correctness:** the substitute `weight` is created with
  `hidden_states.device` / `hidden_states.dtype`, so the subsequent
  `.to(weight.dtype)` is a no-op for the `None` case and does not silently
  promote/demote precision.
- **Tests not added:** the NPU path only runs when `is_torch_npu_available()`
  is true, which is not the case in HF CI. Mocking the guard would not
  exercise the real CANN op and would be fragile. Happy to add a
  device-agnostic test (assert the `weight=None` path matches a reference
  RMSNorm with gamma=1 on CPU) if reviewers want one.
- Intentionally did not touch the `else` branch or bias handling.

## Before submitting

- [x] Did you use an AI agent (Claude Code) to help with this PR?
  - [x] Read the [Coding with AI agents](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution#coding-with-ai-agents) guide.
  - [x] Ran [`self-review`](https://github.com/huggingface/diffusers/blob/main/.ai/skills/self-review/SKILL.md) on the diff (notes above).
  - [x] Shared final self-review notes above.
- [x] Read the [contributor guideline](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution).
- [x] Read the [philosophy doc](https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy) (change is small and localized, bugfix only).
- [ ] Discussed via issue/forum: none yet — opening this PR as the discussion. Can file a tracking issue if preferred.
- [ ] Documentation: no doc changes (internal forward path, no public API change).
- [ ] Tests: not added, see self-review notes.

## Who can review?

NPU doesn't have a dedicated owner in the list; this is a model/normalization
change, so tagging:

@yiyixuxu @dg845

### CODE REVIEW

**Comment by xucqX** [comment]
fixes: #14590

**Comment by xucqX** [comment]
Follow-up to commit `5d489e0`: while resolving the merge conflict with `upstream/main`, the upstream side of the conflict added `and self.weight is not None` to the NPU branch guard. That guard silently reverted this PR's fix — `weight=None` (i.e. `elementwise_affine=False`) was pushed back onto the pure-PyTorch `else` path, losing the fused CANN kernel that this PR exists to provide.
﻿
Commit `46c7553` restores the intended behaviour: on NPU, when `self.weight is None`, we substitute an all-ones tensor so `npu_rms_norm` still gets a valid gamma and stays on the fast path. The upstream `else` branch is otherwise unchanged.
